### DISCUSSION TITLE
Can It work on Multilingual data?

### DISCUSSION THREAD

**Discussion by Komal-99** [imartinez/privateGPT#998] [Q&A]
Hi, 
I want to ask that PrivateGPT only works on English data or can work on French, German dataset?
For ex- We have a corpus of French language, and we made its embeddings using Ingest.py file then my model will be able to create responses in French?

Model using - Llama--7B-chat

**Comment by Lue-C** [comment]
Hi,
I am very interested in building a version of privateGPT that processes non-english texts and gives non-english answers (in my case it would be German). Processing non-english data requires an embedding model, that was trained on the desired language. I tried "distiluse-base-multilingual-cased-v1" and it worked ok (but much better than the default embedding model). Yet there is much room for improvement on the choice.

For getting non-english answers a llm is required which was trained on the desired language. So far I did not find a good overview which models are supported by GPT4All and LLama. Do you know one?
At the moment I am using the "GPT4All-13B-snoozy.ggmlv3.q4_0" model and found that for some prompts it does give German answers and for others it does not. This could be adressed by changing the default prompt.

Did you find a solution in the meantime?

**Comment by Komal-99** [comment]
Actually, PrivateGPT is just a method to use the LLM model, so It depends on which model you are using, I have overcome those language Issues for German by using ``` vicuna 13b ``` ggml model. It works quite well on both English and German text.

**Reply by Lue-C** [reply]
I searched for it but could not find a bin file. Where can I download it?

**Reply by Komal-99** [reply]
https://huggingface.co/TheBloke/Vicuna-13B-v1.3-German-GGML

**Reply by Lue-C** [reply]
Thanks a lot, this works really good:)

**Reply by Komal-99** [reply]
Most welcome , If you cane across any multilingual model please so let me know which can support spanish German french english all.

**Comment by NadaComa** [comment]
Hi, 
so can you confirm that using distiluse-base-multilingual-cased-v1 for embedding and vicuna as LLM gives satisfying results for german documents?

**Reply by Lue-C** [reply]
I can't really say that I am satsified with the results. distiluse-base-multilingual-cased-v1 is the best embedding model I found so far, in 60% of the cases it finds the part of the text (the part containing the answer to the question). Vicuna as LLM is one of the best models I tried so far, but sometimes the answers semantically weird or do not answer the question. As far as I can tell, this is due to the documents processed to give the answer. Another ponint is the choice of the prompt which has a large effect on the result. I am currently working on those points. Do you want to build a bot for german documents too?

**Reply by NadaComa** [reply]
Yes, I do. Interested in sharing experiences in a call?

**Reply by Lue-C** [reply]
Yes, that is a great idea. Which app do you prefer?

**Reply by NadaComa** [reply]
Whatsapp or Telegramm or Signal

**Reply by Lue-C** [reply]
Then let us meet via telegram, you can send me your username to xepml@freenet.de
When do you have time this week?

**Reply by NadaComa** [reply]
Did you receive my mail?

**Comment by Komal-99** [comment]
@Lue-C  which all languages you have tested with distiluse-base-multilingual-cased-v1 ?
Did you found any another for multilingual as vicuna is only for German

**Reply by Lue-C** [reply]
Not yet, but I have not been looking for other models in the last week because of other priorities. I will write if there is progress on that matter.
A possible way around working with several languages could be to detect the language of the prompt and then use a language specific model. Do you think this is feasible?

**Reply by Komal-99** [reply]
Yes. I am also looking over this approach
